test/EtomicSwap.js

```diff
@@ -47,593 +47,663 @@ const invalidSecretHex = '0x' + invalidSecret.toString('hex');
+        expect(tokenBalance).to.equal(BigInt(amountToSend));
 
         // Attempting refund again - should fail
-        await this.swap.senderRefundErc1155(id, amountToSend, secretHash, this.erc1155token.address, tokenId, accounts[1], { from: accounts[0] }).should.be.rejectedWith(EVMThrow);
+        await etomicSwap.connect(accounts[0]).senderRefundErc1155(id, amountToSend, secretHash, erc1155token.target, tokenId, accounts[1].address).should.be.rejected;
     });
 
-    it('should allow receiver to spend ETH payment by revealing a secret', async function () {
+    it('should allow receiver to spend ETH payment by revealing a secret', async function() {
         const lockTime = await currentEvmTime() + 1000;
         const params = [
             id,
-            accounts[1],
+            accounts[1].address,
             secretHash,
             lockTime
         ];
 
-        // should not allow to spend uninitialized payment
-        await this.swap.receiverSpend(id, web3.utils.toWei('1'), secretHex, zeroAddr, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
+        // Should not allow to spend uninitialized payment
+        await etomicSwap.connect(accounts[1]).receiverSpend(id, ethers.parseEther('1'), secretHex, zeroAddr, accounts[0].address).should.be.rejected;
 
-        await this.swap.ethPayment(...params, { value: web3.utils.toWei('1') }).should.be.fulfilled;
+        // Make the ETH payment
+        await etomicSwap.connect(accounts[0]).ethPayment(...params, {
+            value: ethers.parseEther('1')
+        }).should.be.fulfilled;
 
-        // should not allow to spend with invalid secret
-        await this.swap.receiverSpend(id, web3.utils.toWei('1'), invalidSecretHex, zeroAddr, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
-        // should not allow to spend invalid amount
-        await this.swap.receiverSpend(id, web3.utils.toWei('2'), secretHex, zeroAddr, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
+        // Should not allow to spend with invalid secret
+        await etomicSwap.connect(accounts[1]).receiverSpend(id, ethers.parseEther('1'), invalidSecretHex, zeroAddr, accounts[0].address).should.be.rejected;
+        // Should not allow to spend invalid amount
+        await etomicSwap.connect(accounts[1]).receiverSpend(id, ethers.parseEther('2'), secretHex, zeroAddr, accounts[0].address).should.be.rejected;
 
-        // should not allow to claim from non-receiver address even with valid secret
-        await this.swap.receiverSpend(id, web3.utils.toWei('1'), secretHex, zeroAddr, accounts[0], { from: accounts[0] }).should.be.rejectedWith(EVMThrow);
+        // Should not allow to claim from non-receiver address even with valid secret
+        await etomicSwap.connect(accounts[0]).receiverSpend(id, ethers.parseEther('1'), secretHex, zeroAddr, accounts[0].address).should.be.rejected;
```

Comment:
Can we maybe avoid duplication a bit by using the following vars?
```js
let etomicSwapRunner0 = etomicSwap.connect(accounts[0]);
let etomicSwapRunner1 = etomicSwap.connect(accounts[1]);
```

PS, that's actually a nitpick, so final decision is up to you 🙂

Reply:
Nice catch, done